f8a_worker/defaults.py

```diff
@@ -115,11 +116,21 @@ def is_local_deployment(cls):
     @classmethod
     def _rate_limit_exceeded(cls, headers):
         """Return True if Github API rate limit has been exceeded."""
-        # avoid cyclic import
-        from f8a_worker.utils import get_response
-        response = get_response(urljoin(cls.GITHUB_API, "rate_limit"), headers=headers)
-        remaining_attempts = response.get('rate', {}).get('remaining', 0)
-        return remaining_attempts == 0
+        try:
+            response = requests.get(urljoin(cls.GITHUB_API, "rate_limit"),
+                                    headers=headers)
+            response.raise_for_status()
+        except Exception as e:
+            logger.error(e)
+            return True
```

Comment:
It will not abort the ingestion it will not return any token  
https://github.com/jparsai/fabric8-analytics-worker/blob/28e2e51412adc9a76daa6bbd2c1cdf76f4d7b43e/f8a_worker/utils.py#L765
and Github API will be called without any token, Here we may gate response from API or it will throw rate limit issue for and in case of error we set -1 for count   
https://github.com/jparsai/fabric8-analytics-worker/blob/28e2e51412adc9a76daa6bbd2c1cdf76f4d7b43e/f8a_worker/utils.py#L673